File: Tests/AINotebookCoreTests/TabSwitchCoordinatorTests.swift
import XCTest
@testable import AINotebookCore

@MainActor
final class TabSwitchCoordinatorTests: XCTestCase {

    func testRequestSetsTarget() {
        let c = TabSwitchCoordinator()
        XCTAssertNil(c.target)
        c.request(.notes)
        XCTAssertEqual(c.target, .notes)
    }

    func testClearResetsTarget() {
        let c = TabSwitchCoordinator()
        c.request(.notes)
        c.clear()
        XCTAssertNil(c.target)
    }
}

File: windows/tests/AINotebook.App.Tests/TabSwitchCoordinatorTests.cs
using AINotebook.App.Services;
using Xunit;

namespace AINotebook.App.Tests;

public class TabSwitchCoordinatorTests
{
    [Fact]
    public void Request_sets_target()
    {
        var c = new TabSwitchCoordinator();
        Assert.Null(c.Target);
        c.Request(TabSwitchCoordinator.Tab.Notes);
        Assert.Equal(TabSwitchCoordinator.Tab.Notes, c.Target);
    }

    [Fact]
    public void Clear_resets_target()
    {
        var c = new TabSwitchCoordinator();
        c.Request(TabSwitchCoordinator.Tab.Chat);
        c.Clear();
        Assert.Null(c.Target);
    }

    [Fact]
    public void Request_raises_property_changed()
    {
        var c = new TabSwitchCoordinator();
        string? changed = null;
        c.PropertyChanged += (_, e) => changed = e.PropertyName;
        c.Request(TabSwitchCoordinator.Tab.Transformations);
        Assert.Equal(nameof(TabSwitchCoordinator.Target), changed);
    }
}
